cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUpdateProductImagesForFeeds.js

```diff
@@ -103,28 +103,30 @@ function writeProductFileContent(cloudinaryUrlStreamWriter, productSearchHitsItr
                     cloudinaryUrlStreamWriter.writeStartElement('image-group');
                     cloudinaryUrlStreamWriter.writeAttribute('view-type', params.viewType);
 
-                    for (var j = 0; j < imgAssetsSorted.resources.length; j++) {
+                    for (let imgResource of imgAssetsSorted.resources) {
                         cloudinaryUrlStreamWriter.writeStartElement('image');
-                        cloudinaryUrlStreamWriter.writeAttribute('path', imgAssetsSorted.resources[j].public_id);
+                        cloudinaryUrlStreamWriter.writeAttribute('path', imgResource.public_id);
                         cloudinaryUrlStreamWriter.writeEndElement();
                         cloudinaryUrlStreamWriter.writeCharacters('\n');
                     }
                     cloudinaryUrlStreamWriter.writeEndElement();
                     cloudinaryUrlStreamWriter.writeCharacters('\n');
 
                     if (!product.variants.empty) {
-                        for (var k = 0; k < product.variants.length; k++) {
-                            variantsID = product.variants[k].ID;
+                        var variants = product.variants.iterator();
```

Comment:
Why are we avoiding for-of loop here?

Reply:
variants are SFCC collections, according to SFCC best practice we should use iterators on collections and seekable iterators. It is recommended by SFCC